canonical/pytest-interface-tester
# Procedural api

this PR implements the procedural API described in #3 

Fixes #3 

This is how an interface test will look:
```
def test_no_data_on_joined():
    # nothing happens on joined: databags are empty
    t = Tester()
    state = t.run('openfga-relation-joined')
    t.assert_relation_data_empty()
```

where one MUST:
1) instantiate Tester within the scope of the test
2) call t.run
3) call one of 
  a) t.assert_relation_data_empty, or 
  b) t.assert_schema_valid or 
  c) t.skip_schema_validation

in exactly this order. Otherwise the test will fail.

also, one CAN:
- write assertions on `state`, to verify it matches the expectations

tandem PR: https://github.com/canonical/charm-relation-interfaces/pull/101
